src/codeu/chat/client/simplegui/originalgui/MessagePanel.java

```diff
@@ -0,0 +1,190 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package codeu.chat.client.simplegui.originalgui;
+
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import javax.swing.*;
+
+import codeu.chat.client.ClientContext;
+import codeu.chat.common.ConversationSummary;
+import codeu.chat.common.Message;
+import codeu.chat.common.User;
+
+// NOTE: JPanel is serializable, but there is no need to serialize MessagePanel
+// without the @SuppressWarnings, the compiler will complain of no override for serialVersionUID
+@SuppressWarnings("serial")
+public final class MessagePanel extends JPanel {
+
+  // These objects are modified by the Conversation Panel.
+  private final JLabel messageOwnerLabel = new JLabel("Owner:", JLabel.RIGHT);
+  private final JLabel messageConversationLabel = new JLabel("Conversation:", JLabel.LEFT);
+  private final DefaultListModel<String> messageListModel = new DefaultListModel<>();
+
+  private final ClientContext clientContext;
+
+  public MessagePanel(ClientContext clientContext) {
+    super(new GridBagLayout());
+    this.clientContext = clientContext;
+    initialize();
+  }
+
+  // External agent calls this to trigger an update of this panel's contents.
+  public void update(ConversationSummary owningConversation) {
+
+    final User u = (owningConversation == null) ?
```

Comment:
nit (google style): ? goes on next line, as does the : below

i.e.:
(owningConversation == null)
? null
: clientContext...